unittest/client/src/ilm_control_unittests.cpp

```diff
@@ -0,0 +1,5071 @@
+ *                         +# wl_display_roundtrip_queue() not be called
+ */
+TEST_F(IlmControlTest, ilm_surfaceGetOpacity_wrongVisibility)
+{
+    t_ilm_surface l_surfaceId = 1;
+    t_ilm_float *p_opacity = nullptr;
+    ASSERT_EQ(ILM_FAILED, ilm_surfaceGetOpacity(l_surfaceId, p_opacity));
+
+    ASSERT_EQ(0, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(0, wl_display_roundtrip_queue_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_surfaceGetOpacity_cannotGetRoundTripQueue
+ * @brief               Test case of ilm_surfaceGetOpacity() where wl_display_roundtrip_queue() fails, return -1
+ *                      and ilm_context wl.controller does return null object
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_display_roundtrip_queue() does return -1
+ *                      -# Calling the ilm_surfaceGetOpacity() time 1 with ilm_context wl.controller doesn't return null object
+ *                      -# Calling the ilm_surfaceGetOpacity() time 2 with ilm_context wl.controller does return null object
+ *                      -# Verification point:
+ *                         +# Both of ilm_surfaceGetOpacity() time 1 and time 2 must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags() must be called
+ *                         +# wl_display_roundtrip_queue() must be called
+ */
+TEST_F(IlmControlTest, ilm_surfaceGetOpacity_cannotGetRoundTripQueue)
+{
+    SET_RETURN_SEQ(wl_display_roundtrip_queue, mp_failureResult, 1);
+
+    t_ilm_surface l_surfaceId = 1;
+    t_ilm_float l_opacity = 1;
+    ASSERT_EQ(ILM_FAILED, ilm_surfaceGetOpacity(l_surfaceId, &l_opacity));
+
+    ASSERT_EQ(1, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(1, wl_display_roundtrip_queue_fake.call_count);
+
+    ilm_context.wl.controller = nullptr;
+    ASSERT_EQ(ILM_FAILED, ilm_surfaceGetOpacity(l_surfaceId, &l_opacity));
+
+    ASSERT_EQ(2, wl_proxy_marshal_flags_fake.call_count);
+    ASSERT_EQ(2, wl_display_roundtrip_queue_fake.call_count);
+}
+
+/** ================================================================================================
+ * @test_id             ilm_surfaceGetOpacity_invalidLayerId
+ * @brief               Test case of ilm_surfaceGetOpacity() where invalid layer id
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_display_roundtrip_queue() does return 0 (success)
+ *                      -# Calling the ilm_surfaceGetOpacity() with invalid layer id
+ *                      -# Verification point:
+ *                         +# ilm_surfaceGetOpacity() must return ILM_FAILED
+ *                         +# wl_proxy_marshal_flags() must be called once time
+ *                         +# wl_display_roundtrip_queue() must be called once time
+ */
+TEST_F(IlmControlTest, ilm_surfaceGetOpacity_invalidLayerId)
```

Comment:
Here I think we are checking for invalid surface id name and description should be accordingly